### ISSUE TITLE
Workspace status command on Windows produces inconsistently ^M characters

### ISSUE THREAD

**Issue by facewindu** [bazelbuild/bazel#13442]
### Description of the problem / feature request:

Workspace status command on Windows produces weird ^M character

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create a `foo.cmd` file in your bazel workspace with the content
```
@echo off
echo BLA foo^&foo2
echo BLA2 foo3^&foo4
```

Run any target (like eg the [java tutorial one](https://docs.bazel.build/versions/master/tutorial/java.html))
`bazel build --workspace_status_command=foo.cmd //:ProjectRunner`

Check the produced `bazel-out/volatile-status.txt`:
```
BLA foo&foo2^M
BLA2 foo3&foo4
BUILD_TIMESTAMP 1620395339
```

### What operating system are you running Bazel on?

Windows 10
Version 1909 (OS Build 18363.1500)

### What's the output of `bazel info release`?

release 4.0.0

###  Have you found anything relevant by searching the web?

No

**Comment by iancha1992** [comment]
@bazel-io fork 8.0.0

**Comment by Wyverald** [comment]
@iancha1992 This is already fixed by #23922, right? We just need to cherry-pick it to release-8.0.0.

**Comment by iancha1992** [comment]
@Wyverald https://github.com/bazelbuild/bazel/pull/24324

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.0 RC3](https://github.com/bazelbuild/bazel/releases/tag/8.0.0rc3). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.0rc3. Thanks!
